plugin/src/index.ts

```diff
@@ -47,6 +47,28 @@ export async function onPreBuild({
   await checkNetlifyImageCdn({ netlifyConfig })
 }
 
+export async function onDev({
+  netlifyConfig,
+  constants,
+}: NetlifyPluginOptions): Promise<void> {
+  // eslint-disable-next-line no-param-reassign
+  netlifyConfig.build.environment.GATSBY_PRECOMPILE_DEVELOP_FUNCTIONS = `true`
```

Comment:
Gatsby compiles functions lazily in dev by default which cause problems if we want to wrap them - this will force compile them eagerly